Closes #110 | Add WikiMatrix Dataloader

yana-xuyan opened the pull request:
Closes #110 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

ljvmiranda921 commented on seacrowd/sea_datasets/wikimatrix/wikimatrix.py:
```diff
@@ -0,0 +1,156 @@
+from typing import List
+
+import datasets
+import json
```
```suggestion
```

ljvmiranda921 commented on seacrowd/sea_datasets/wikimatrix/wikimatrix.py:
```diff
@@ -0,0 +1,156 @@
+    url = "https://aclanthology.org/2021.eacl-main.115",
+    doi = "10.18653/v1/2021.eacl-main.115",
+    pages = "1351--1361",
+    abstract = "We present an approach based on multilingual sentence embeddings to automatically extract parallel sentences from the content 
+    of Wikipedia articles in 96 languages, including several dialects or low-resource languages. We do not limit the extraction process to 
+    alignments with English, but we systematically consider all possible language pairs. In total, we are able to extract 135M parallel sentences 
+    for 16720 different language pairs, out of which only 34M are aligned with English. This corpus is freely available. To get an indication 
+    on the quality of the extracted bitexts, we train neural MT baseline systems on the mined data only for 1886 languages pairs, and evaluate 
+    them on the TED corpus, achieving strong BLEU scores for many language pairs. The WikiMatrix bitexts seem to be particularly interesting 
+    to train MT systems between distant languages without the need to pivot through English.",
```
```suggestion
    abstract = "We present an approach based on multilingual sentence embeddings to automatically extract parallel sentences from the content
    of Wikipedia articles in 96 languages, including several dialects or low-resource languages. We do not limit the extraction process to
    alignments with English, but we systematically consider all possible language pairs. In total, we are able to extract 135M parallel sentences
    for 16720 different language pairs, out of which only 34M are aligned with English. This corpus is freely available. To get an indication
    on the quality of the extracted bitexts, we train neural MT baseline systems on the mined data only for 1886 languages pairs, and evaluate
    them on the TED corpus, achieving strong BLEU scores for many language pairs. The WikiMatrix bitexts seem to be particularly interesting
    to train MT systems between distant languages without the need to pivot through English.",
```

ljvmiranda921 commented on seacrowd/sea_datasets/wikimatrix/wikimatrix.py:
```diff
@@ -0,0 +1,156 @@
+
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+_SUBSETS = ["en-jv", "es-jv", "fr-jv", "id-jv", "it-jv", "jv-pt", "bg-ceb", "ar-ceb", "ca-ceb", "ceb-cs", "ceb-de", "ceb-en", "ceb-es", "ceb-fi", "ceb-fr", "ceb-hu", "ceb-it", "ceb-ja", "ceb-nl", "ceb-no", "ceb-pl", "ceb-pt", "ceb-ro", "ceb-ru", "ceb-sv", "ceb-uk", "id-is", "id-it", "id-ja", "id-ko", "id-lt", "id-mk", "id-ml", "id-mr", "id-ne", "id-nl", "id-no", "id-pl", "id-pt", "id-ro", "id-ru" ...
+
```
Please rerun the `make check_file=...` script after making this change. Here, I skipped formatting the list because it formats everything with one element per line.

```suggestion
# fmt: off
_SUBSETS = ["en-jv", "es-jv", "fr-jv", "id-jv", "it-jv", "jv-pt", "bg-ceb", "ar-ceb", "ca-ceb", "ceb-cs", "ceb-de", "ceb-en", "ceb-es", "ceb-fi", "ceb-fr", "ceb-hu", "ceb-it", "ceb-ja", "ceb-nl", "ceb-no", "ceb-pl", "ceb-pt", "ceb-ro", "ceb-ru", "ceb-sv", "ceb-uk", "id-is", "id-it", "id-ja", "id-ko", "id-lt", "id-mk", "id-ml", "id-mr", "id-ne", "id-nl", "id-no", "id-pl", "id-pt", "id-ro", "id-ru", "id-sh", "id-si", "id-sk", "id-sl", "id-sq", "id-sr", "id-sv", "id-sw", "id-ta", "id-te", "id-tl", "id-tr", "id-tt", "id-uk", "id-vi", "id-zh", "ar-id", "az-id", "ba-id", "bg-id", "bn-id", "bs-id", "ca-id", "cs-id", "da-id", "de-id", "el-id", "en-id", "eo-id", "es-id", "et-id", "eu-id", "fa-id", "fi-id", "fr-id", "gl-id", "he-id", "hi-id", "hr-id", "hu-id", "ar-tl", "bg-tl", "bs-tl", "ca-tl", "cs-tl", "da-tl", "de-tl", "el-tl", "en-tl", "eo-tl", "es-tl", "et-tl", "fi-tl", "fr-tl", "gl-tl", "he-tl", "hr-tl", "hu-tl", "it-tl", "ja-tl", "lt-tl", "mk-tl", "nl-tl", "no-tl", "pl-tl", "pt-tl", "ro-tl", "ru-tl", "sh-tl", "sk-tl", "sl-tl", "sq-tl", "sr-tl", "sv-tl", "tl-tr", "tl-uk", "tl-vi", "tl-zh", "ar-vi", "az-vi", "bg-vi", "bn-vi", "bs-vi", "ca-vi", "cs-vi", "da-vi", "de-vi", "el-vi", "en-vi", "eo-vi", "es-vi", "et-vi", "eu-vi", "fa-vi", "fi-vi", "fr-vi", "gl-vi", "he-vi", "hi-vi", "hr-vi", "hu-vi", "is-vi", "it-vi", "ja-vi", "ko-vi", "lt-vi", "mk-vi", "ml-vi", "mr-vi", "nl-vi", "no-vi", "pl-vi", "pt-vi", "ro-vi", "ru-vi", "sh-vi", "si-vi", "sk-vi", "sl-vi", "sq-vi", "sr-vi", "sv-vi", "sw-vi", "ta-vi", "te-vi", "tr-vi", "uk-vi", "vi-zh"] 
# fmt: on
```

ljvmiranda921 commented on seacrowd/sea_datasets/wikimatrix/wikimatrix.py:
```diff
@@ -0,0 +1,156 @@
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        lang1, lang2 = self.config.name.split("_")[1], self.config.name.split("_")[2]
+        filepath = Path(dl_manager.download_and_extract(_URLs[_DATASETNAME].replace("{lang1}", lang1).replace("{lang2}", lang2)))
```
I don't mind this approach, but maybe you can use string formatting (`.format`) here? It seems more idiomatic for this use-case.

ljvmiranda921 commented on seacrowd/sea_datasets/wikimatrix/wikimatrix.py:
```diff
@@ -0,0 +1,156 @@
+
+_LICENSE = Licenses.CC_BY_SA_4_0.value
+
+_URLs = {_DATASETNAME: "https://dl.fbaipublicfiles.com/laser/WikiMatrix/v1/WikiMatrix.{lang1}-{lang2}.tsv.gz"}
```
You can use `_URL = "https://..."` for simplicity

ljvmiranda921 reviewed (commented):
Hi @yana-xuyan , thanks for this PR!
I have a few nits mostly on formatting, but the dataloader works as expected. Note that I only tested a few pairs (randomly selected) that my machine can accommodate.

yana-xuyan commented:
Hi Miranda,

Thank you for your comments! I’ll modify accordingly. :)

Best regards,
Yan

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Lj Miranda ***@***.***>
发送时间: Thursday, January 4, 2024 8:52:19 AM
收件人: SEACrowd/seacrowd-datahub ***@***.***>
抄送: Yan XU ***@***.***>; Mention ***@***.***>
主题: Re: [SEACrowd/seacrowd-datahub] Closes #110 | Add WikiMatrix Dataloader (PR #244)


@ljvmiranda921 commented on this pull request.

Hi @yana-xuyan<https://github.com/yana-xuyan> , thanks for this PR!
I have a few nits mostly on formatting, but the dataloader works as expected. Note that I only tested a few pairs (randomly selected) that my machine can accommodate.

________________________________

In seacrowd/sea_datasets/wikimatrix/wikimatrix.py<https://github.com/SEACrowd/seacrowd-datahub/pull/244#discussion_r1441155140>:

> @@ -0,0 +1,156 @@
+from pathlib import Path
+from typing import List
+
+import datasets
+import json


⬇️ Suggested change

-import json


________________________________

In seacrowd/sea_datasets/wikimatrix/wikimatrix.py<https://github.com/SEACrowd/seacrowd-datahub/pull/244#discussion_r1441155951>:

> +    abstract = "We present an approach based on multilingual sentence embeddings to automatically extract parallel sentences from the content
+    of Wikipedia articles in 96 languages, including several dialects or low-resource languages. We do not limit the extraction process to
+    alignments with English, but we systematically consider all possible language pairs. In total, we are able to extract 135M parallel sentences
+    for 16720 different language pairs, out of which only 34M are aligned with English. This corpus is freely available. To get an indication
+    on the quality of the extracted bitexts, we train neural MT baseline systems on the mined data only for 1886 languages pairs, and evaluate
+    them on the TED corpus, achieving strong BLEU scores for many language pairs. The WikiMatrix bitexts seem to be particularly interesting
+    to train MT systems between distant languages without the need to pivot through English.",


⬇️ Suggested change

-    abstract = "We present an approach based on multilingual sentence embeddings to automatically extract parallel sentences from the content
-    of Wikipedia articles in 96 languages, including several dialects or low-resource languages. We do not limit the extraction process to
-    alignments with English, but we systematically consider all possible language pairs. In total, we are able to extract 135M parallel sentences
-    for 16720 different language pairs, out of which only 34M are aligned with English. This corpus is freely available. To get an indication
-    on the quality of the extracted bitexts, we train neural MT baseline systems on the mined data only for 1886 languages pairs, and evaluate
-    them on the TED corpus, achieving strong BLEU scores for many language pairs. The WikiMatrix bitexts seem to be particularly interesting
-    to train MT systems between distant languages without the need to pivot through English.",
+    abstract = "We present an approach based on multilingual sentence embeddings to automatically extract parallel sentences from the content
+    of Wikipedia articles in 96 languages, including several dialects or low-resource languages. We do not limit the extraction process to
+    alignments with English, but we systematically consider all possible language pairs. In total, we are able to extract 135M parallel sentences
+    for 16720 different language pairs, out of which only 34M are aligned with English. This corpus is freely available. To get an indication
+    on the quality of the extracted bitexts, we train neural MT baseline systems on the mined data only for 1886 languages pairs, and evaluate
+    them on the TED corpus, achieving strong BLEU scores for many language pairs. The WikiMatrix bitexts seem to be particularly interesting
+    to train MT systems between distant languages without the need to pivot through English.",


________________________________

In seacrowd/sea_datasets/wikimatrix/wikimatrix.py<https://github.com/SEACrowd/seacrowd-datahub/pull/244#discussion_r1441157182>:

> +
+_SUBSETS = ["en-jv", "es-jv", "fr-jv", "id-jv", "it-jv", "jv-pt", "bg-ceb", "ar-ceb", "ca-ceb", "ceb-cs", "ceb-de", "ceb-en", "ceb-es", "ceb-fi", "ceb-fr", "ceb-hu", "ceb-it", "ceb-ja", "ceb-nl", "ceb-no", "ceb-pl", "ceb-pt", "ceb-ro", "ceb-ru", "ceb-sv", "ceb-uk", "id-is", "id-it", "id-ja", "id-ko", "id-lt", "id-mk", "id-ml", "id-mr", "id-ne", "id-nl", "id-no", "id-pl", "id-pt", "id-ro", "id-ru", "id-sh", "id-si", "id-sk", "id-sl", "id-sq", "id-sr", "id-sv", "id-sw", "id-ta", "id-te", "id-tl", "id-tr", "id-tt", "id-uk", "id-vi", "id-zh", "ar-id", "az-id", "ba-id", "bg-id", "bn-id", "bs-id", "ca-id", "cs-id", "da-id", "de-id", "el-id", "en-id", "eo-id", "es-id", "et-id", "eu-id", "fa-id", "fi-id", "fr-id", "gl-id", "he-id", "hi-id", "hr-id", "hu-id", "ar-tl", "bg-tl", "bs-tl", "ca-tl", "cs-tl", "da-tl", "de-tl", "el-tl", "en-tl", "eo-tl", "es-tl", "et-tl", "fi-tl", "fr-tl", "gl-tl", "he-tl", "hr-tl", "hu-tl", "it-tl", "ja-tl", "lt-tl", "mk-tl", "nl-tl", "no-tl", "pl-tl", "pt-tl", "ro-tl", "ru-tl", "sh-tl", "sk-tl", "sl-tl", "sq-tl", "sr-tl", "sv-tl", "tl-tr", "tl-uk", "tl-vi", "tl-zh", "ar-vi", "az-vi", "bg-vi", "bn-vi", "bs-vi", "ca-vi", "cs-vi", "da-vi", "de-vi", "el-vi", "en-vi", "eo-vi", "es-vi", "et-vi", "eu-vi", "fa-vi", "fi-vi", "fr-vi", "gl-vi", "he-vi", "hi-vi", "hr-vi", "hu-vi", "is-vi", "it-vi", "ja-vi", "ko-vi", "lt-vi", "mk-vi", "ml-vi", "mr-vi", "nl-vi", "no-vi", "pl-vi", "pt-vi", "ro-vi", "ru-vi", "sh-vi", "si-vi", "sk-vi", "sl-vi", "sq-vi", "sr-vi", "sv-vi", "sw-vi", "ta-vi", "te-vi", "tr-vi", "uk-vi", "vi-zh"]
+


Please rerun the make check_file=... script after making this change. Here, I skipped formatting the list because it formats everything with one element per line.

⬇️ Suggested change

-
-_SUBSETS = ["en-jv", "es-jv", "fr-jv", "id-jv", "it-jv", "jv-pt", "bg-ceb", "ar-ceb", "ca-ceb", "ceb-cs", "ceb-de", "ceb-en", "ceb-es", "ceb-fi", "ceb-fr", "ceb-hu", "ceb-it", "ceb-ja", "ceb-nl", "ceb-no", "ceb-pl", "ceb-pt", "ceb-ro", "ceb-ru", "ceb-sv", "ceb-uk", "id-is", "id-it", "id-ja", "id-ko", "id-lt", "id-mk", "id-ml", "id-mr", "id-ne", "id-nl", "id-no", "id-pl", "id-pt", "id-ro", "id-ru", "id-sh", "id-si", "id-sk", "id-sl", "id-sq", "id-sr", "id-sv", "id-sw", "id-ta", "id-te", "id-tl", "id-tr", "id-tt", "id-uk", "id-vi", "id-zh", "ar-id", "az-id", "ba-id", "bg-id", "bn-id", "bs-id", "ca-id", "cs-id", "da-id", "de-id", "el-id", "en-id", "eo-id", "es-id", "et-id", "eu-id", "fa-id", "fi-id", "fr-id", "gl-id", "he-id", "hi-id", "hr-id", "hu-id", "ar-tl", "bg-tl", "bs-tl", "ca-tl", "cs-tl", "da-tl", "de-tl", "el-tl", "en-tl", "eo-tl", "es-tl", "et-tl", "fi-tl", "fr-tl", "gl-tl", "he-tl", "hr-tl", "hu-tl", "it-tl", "ja-tl", "lt-tl", "mk-tl", "nl-tl", "no-tl", "pl-tl", "pt-tl", "ro-tl", "ru-tl", "sh-tl", "sk-tl", "sl-tl", "sq-tl", "sr-tl", "sv-tl", "tl-tr", "tl-uk", "tl-vi", "tl-zh", "ar-vi", "az-vi", "bg-vi", "bn-vi", "bs-vi", "ca-vi", "cs-vi", "da-vi", "de-vi", "el-vi", "en-vi", "eo-vi", "es-vi", "et-vi", "eu-vi", "fa-vi", "fi-vi", "fr-vi", "gl-vi", "he-vi", "hi-vi", "hr-vi", "hu-vi", "is-vi", "it-vi", "ja-vi", "ko-vi", "lt-vi", "mk-vi", "ml-vi", "mr-vi", "nl-vi", "no-vi", "pl-vi", "pt-vi", "ro-vi", "ru-vi", "sh-vi", "si-vi", "sk-vi", "sl-vi", "sq-vi", "sr-vi", "sv-vi", "sw-vi", "ta-vi", "te-vi", "tr-vi", "uk-vi", "vi-zh"]
-
+# fmt: off
+_SUBSETS = ["en-jv", "es-jv", "fr-jv", "id-jv", "it-jv", "jv-pt", "bg-ceb", "ar-ceb", "ca-ceb", "ceb-cs", "ceb-de", "ceb-en", "ceb-es", "ceb-fi", "ceb-fr", "ceb-hu", "ceb-it", "ceb-ja", "ceb-nl", "ceb-no", "ceb-pl", "ceb-pt", "ceb-ro", "ceb-ru", "ceb-sv", "ceb-uk", "id-is", "id-it", "id-ja", "id-ko", "id-lt", "id-mk", "id-ml", "id-mr", "id-ne", "id-nl", "id-no", "id-pl", "id-pt", "id-ro", "id-ru", "id-sh", "id-si", "id-sk", "id-sl", "id-sq", "id-sr", "id-sv", "id-sw", "id-ta", "id-te", "id-tl", "id-tr", "id-tt", "id-uk", "id-vi", "id-zh", "ar-id", "az-id", "ba-id", "bg-id", "bn-id", "bs-id", "ca-id", "cs-id", "da-id", "de-id", "el-id", "en-id", "eo-id", "es-id", "et-id", "eu-id", "fa-id", "fi-id", "fr-id", "gl-id", "he-id", "hi-id", "hr-id", "hu-id", "ar-tl", "bg-tl", "bs-tl", "ca-tl", "cs-tl", "da-tl", "de-tl", "el-tl", "en-tl", "eo-tl", "es-tl", "et-tl", "fi-tl", "fr-tl", "gl-tl", "he-tl", "hr-tl", "hu-tl", "it-tl", "ja-tl", "lt-tl", "mk-tl", "nl-tl", "no-tl", "pl-tl", "pt-tl", "ro-tl", "ru-tl", "sh-tl", "sk-tl", "sl-tl", "sq-tl", "sr-tl", "sv-tl", "tl-tr", "tl-uk", "tl-vi", "tl-zh", "ar-vi", "az-vi", "bg-vi", "bn-vi", "bs-vi", "ca-vi", "cs-vi", "da-vi", "de-vi", "el-vi", "en-vi", "eo-vi", "es-vi", "et-vi", "eu-vi", "fa-vi", "fi-vi", "fr-vi", "gl-vi", "he-vi", "hi-vi", "hr-vi", "hu-vi", "is-vi", "it-vi", "ja-vi", "ko-vi", "lt-vi", "mk-vi", "ml-vi", "mr-vi", "nl-vi", "no-vi", "pl-vi", "pt-vi", "ro-vi", "ru-vi", "sh-vi", "si-vi", "sk-vi", "sl-vi", "sq-vi", "sr-vi", "sv-vi", "sw-vi", "ta-vi", "te-vi", "tr-vi", "uk-vi", "vi-zh"]
+# fmt: on


________________________________

In seacrowd/sea_datasets/wikimatrix/wikimatrix.py<https://github.com/SEACrowd/seacrowd-datahub/pull/244#discussion_r1441157895>:

> +                "text_2_name": datasets.Value("string"),
+            })
+        elif self.config.schema == "seacrowd_t2t":
+            features = schemas.text2text_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        lang1, lang2 = self.config.name.split("_")[1], self.config.name.split("_")[2]
+        filepath = Path(dl_manager.download_and_extract(_URLs[_DATASETNAME].replace("{lang1}", lang1).replace("{lang2}", lang2)))


I don't mind this approach, but maybe you can use string formatting (.format) here? It seems more idiomatic for this use-case.

________________________________

In seacrowd/sea_datasets/wikimatrix/wikimatrix.py<https://github.com/SEACrowd/seacrowd-datahub/pull/244#discussion_r1441158271>:

> +    them on the TED corpus, achieving strong BLEU scores for many language pairs. The WikiMatrix bitexts seem to be particularly interesting
+    to train MT systems between distant languages without the need to pivot through English.",
+}
+"""
+
+_DESCRIPTION = """\
+WikiMatrix is automatically extracted parallel sentences from the content of Wikipedia articles in 96 languages, including several dialects
+or low-resource languages. 8 languages among them are spoken in Southeast Asia region. In total, there are 135M parallel sentences from 1620
+different language pairs.
+"""
+
+_HOMEPAGE = "https://github.com/facebookresearch/LASER/tree/main/tasks/WikiMatrix"
+
+_LICENSE = Licenses.CC_BY_SA_4_0.value
+
+_URLs = {_DATASETNAME: "https://dl.fbaipublicfiles.com/laser/WikiMatrix/v1/WikiMatrix.{lang1}-{lang2}.tsv.gz"}


You can use _URL = "https://..." for simplicity

—
Reply to this email directly, view it on GitHub<https://github.com/SEACrowd/seacrowd-datahub/pull/244#pullrequestreview-1803287325>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJGALO2GJPYBYAECVLD6ZRLYMX4MHAVCNFSM6AAAAABBEHZRRGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTQMBTGI4DOMZSGU>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

ljvmiranda921 commented on seacrowd/sea_datasets/wikimatrix/wikimatrix.py:
```diff
@@ -0,0 +1,156 @@
+_SUBSETS = ["en-jv", "es-jv", "fr-jv", "id-jv", "it-jv", "jv-pt", "bg-ceb", "ar-ceb", "ca-ceb", "ceb-cs", "ceb-de", "ceb-en", "ceb-es", "ceb-fi", "ceb-fr", "ceb-hu", "ceb-it", "ceb-ja", "ceb-nl", "ceb-no", "ceb-pl", "ceb-pt", "ceb-ro", "ceb-ru", "ceb-sv", "ceb-uk", "id-is", "id-it", "id-ja", "id-ko", "id-lt", "id-mk", "id-ml", "id-mr", "id-ne", "id-nl", "id-no", "id-pl", "id-pt", "id-ro", "id-ru" ...
+
+
+class WikiMatrix(datasets.GeneratorBasedBuilder):
```
```suggestion
class WikiMatrixDataset(datasets.GeneratorBasedBuilder):
```

sabilmakbar commented on seacrowd/sea_datasets/wikimatrix/wikimatrix.py:
```diff
@@ -0,0 +1,156 @@
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+_SUBSETS = ["en-jv", "es-jv", "fr-jv", "id-jv", "it-jv", "jv-pt", "bg-ceb", "ar-ceb", "ca-ceb", "ceb-cs", "ceb-de", "ceb-en", "ceb-es", "ceb-fi", "ceb-fr", "ceb-hu", "ceb-it", "ceb-ja", "ceb-nl", "ceb-no", "ceb-pl", "ceb-pt", "ceb-ro", "ceb-ru", "ceb-sv", "ceb-uk", "id-is", "id-it", "id-ja", "id-ko", "id-lt", "id-mk", "id-ml", "id-mr", "id-ne", "id-nl", "id-no", "id-pl", "id-pt", "id-ro", "id-ru" ...
```
Hi @yana-xuyan, can you limit the subsets to only languages defined in `_LANGUAGES` list? thx!

also this should apply to your other PR-ed dataloaders (if you happen to implement it for all langs)

I happen to see this, but the complete review process will be done by LJ & Samcah

ljvmiranda921 replied:
Ah I see, yeah agree with Sabil, let's limit the language combinations to SEA languages.

SamuelCahyawijaya replied:
Hello, @yana-xuyan  @sabilmakbar  @ljvmiranda921, I discussed with @holylovenia regarding the convention for the language pair for parallel data, and we decided to take the other language pairs for parallel data, as it is also an important use case in SEA (translating SEA language from/to language other than SEA).

So, in this case, we can keep the existing subsets as long as one of the languages in the language pair is an SEA language.

yana-xuyan commented:
Hi hi, I’m sorry, but I have already limited the subsets to only languages defined in the _LANGUAGES list. Originally there are 1680 language pairs. Or do you mean I should only keep the language pair if both languages are on the list?

Yan

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Salsabil Maulana Akbar ***@***.***>
发送时间: Thursday, January 4, 2024 7:18:48 PM
收件人: SEACrowd/seacrowd-datahub ***@***.***>
抄送: Yan XU ***@***.***>; Mention ***@***.***>
主题: Re: [SEACrowd/seacrowd-datahub] Closes #110 | Add WikiMatrix Dataloader (PR #244)


@sabilmakbar commented on this pull request.

________________________________

In seacrowd/sea_datasets/wikimatrix/wikimatrix.py<https://github.com/SEACrowd/seacrowd-datahub/pull/244#discussion_r1441628348>:

> +or low-resource languages. 8 languages among them are spoken in Southeast Asia region. In total, there are 135M parallel sentences from 1620
+different language pairs.
+"""
+
+_HOMEPAGE = "https://github.com/facebookresearch/LASER/tree/main/tasks/WikiMatrix"
+
+_LICENSE = Licenses.CC_BY_SA_4_0.value
+
+_URLs = {_DATASETNAME: "https://dl.fbaipublicfiles.com/laser/WikiMatrix/v1/WikiMatrix.{lang1}-{lang2}.tsv.gz"}
+
+_SUPPORTED_TASKS = [Tasks.MACHINE_TRANSLATION]
+
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+_SUBSETS = ["en-jv", "es-jv", "fr-jv", "id-jv", "it-jv", "jv-pt", "bg-ceb", "ar-ceb", "ca-ceb", "ceb-cs", "ceb-de", "ceb-en", "ceb-es", "ceb-fi", "ceb-fr", "ceb-hu", "ceb-it", "ceb-ja", "ceb-nl", "ceb-no", "ceb-pl", "ceb-pt", "ceb-ro", "ceb-ru", "ceb-sv", "ceb-uk", "id-is", "id-it", "id-ja", "id-ko", "id-lt", "id-mk", "id-ml", "id-mr", "id-ne", "id-nl", "id-no", "id-pl", "id-pt", "id-ro", "id-ru", "id-sh", "id-si", "id-sk", "id-sl", "id-sq", "id-sr", "id-sv", "id-sw", "id-ta", "id-te", "id-tl", "id-tr", "id-tt", "id-uk", "id-vi", "id-zh", "ar-id", "az-id", "ba-id", "bg-id", "bn-id", "bs-id", "ca-id", "cs-id", "da-id", "de-id", "el-id", "en-id", "eo-id", "es-id", "et-id", "eu-id", "fa-id", "fi-id", "fr-id", "gl-id", "he-id", "hi-id", "hr-id", "hu-id", "ar-tl", "bg-tl", "bs-tl", "ca-tl", "cs-tl", "da-tl", "de-tl", "el-tl", "en-tl", "eo-tl", "es-tl", "et-tl", "fi-tl", "fr-tl", "gl-tl", "he-tl", "hr-tl", "hu-tl", "it-tl", "ja-tl", "lt-tl", "mk-tl", "nl-tl", "no-tl", "pl-tl", "pt-tl", "ro-tl", "ru-tl", "sh-tl", "sk-tl", "sl-tl", "sq-tl", "sr-tl", "sv-tl", "tl-tr", "tl-uk", "tl-vi", "tl-zh", "ar-vi", "az-vi", "bg-vi", "bn-vi", "bs-vi", "ca-vi", "cs-vi", "da-vi", "de-vi", "el-vi", "en-vi", "eo-vi", "es-vi", "et-vi", "eu-vi", "fa-vi", "fi-vi", "fr-vi", "gl-vi", "he-vi", "hi-vi", "hr-vi", "hu-vi", "is-vi", "it-vi", "ja-vi", "ko-vi", "lt-vi", "mk-vi", "ml-vi", "mr-vi", "nl-vi", "no-vi", "pl-vi", "pt-vi", "ro-vi", "ru-vi", "sh-vi", "si-vi", "sk-vi", "sl-vi", "sq-vi", "sr-vi", "sv-vi", "sw-vi", "ta-vi", "te-vi", "tr-vi", "uk-vi", "vi-zh"]


Hi @yana-xuyan<https://github.com/yana-xuyan>, can you limit the subsets to only languages defined in _LANGUAGES list? thx!

also this should apply to your other PR-ed dataloaders (if you happen to implement it for all langs)

I happen to see this, but the complete review process will be done by LJ & Samcah

―
Reply to this email directly, view it on GitHub<https://github.com/SEACrowd/seacrowd-datahub/pull/244#pullrequestreview-1803961636>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJGALO3ZZ4XEUZQCZGFP2XTYM2FZRAVCNFSM6AAAAABBEHZRRGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTQMBTHE3DCNRTGY>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

sabilmakbar commented:
> Hi hi, I’m sorry, but I have already limited the subsets to only languages defined in the _LANGUAGES list. Originally there are 1680 language pairs. Or do you mean I should only keep the language pair if both languages are on the list?
> 
> Yan

Ugh, apologies for not noticing that. In that case, to make it readable, can you make a dict of eligible languages in SEA with its value of eligible pairs on that dataset. It shd be look like this:

```python
config = {
"mya": ["ind", "eng", "ceb", "tha"],
"ind": ["spa", "mya", "ceb", "msa"]}
```

and construct the `_SUBSET` list according to the dict items? so that the readability will be improved greatly

yana-xuyan commented:
Sure!

Yan

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Salsabil Maulana Akbar ***@***.***>
发送时间: Thursday, January 4, 2024 7:46:12 PM
收件人: SEACrowd/seacrowd-datahub ***@***.***>
抄送: Yan XU ***@***.***>; Mention ***@***.***>
主题: Re: [SEACrowd/seacrowd-datahub] Closes #110 | Add WikiMatrix Dataloader (PR #244)


Hi hi, I’m sorry, but I have already limited the subsets to only languages defined in the _LANGUAGES list. Originally there are 1680 language pairs. Or do you mean I should only keep the language pair if both languages are on the list?

Yan

Ugh, apologies for not noticing that. In that case, to make it readable, can you make a dict of eligible languages in SEA with its value of eligible pairs on that dataset. It shd be look like this:

config = {
"mya": ["ind", "eng", "ceb", "tha"],
"ind": ["spa", "mya", "ceb", "msa"]}

and construct the _SUBSET list according to the dict items? so that the readability will be improved greatly

―
Reply to this email directly, view it on GitHub<https://github.com/SEACrowd/seacrowd-datahub/pull/244#issuecomment-1876963582>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AJGALO7WWB46XKRGJ3VF2F3YM2JAJAVCNFSM6AAAAABBEHZRRGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNZWHE3DGNJYGI>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

ljvmiranda921 commented:
Hi @yana-xuyan just let us know if this is ready for another round of reviews! Thank you! 🙇

SamuelCahyawijaya reviewed (approved):
LGTM!!

yana-xuyan commented:
Hi, @SamuelCahyawijaya @sabilmakbar @ljvmiranda921, I have modified how we initialize the `_SUBSETS` based on @sabilmakbar's suggestion and re-run `make check_file=xxx`. All the previous comments are resolved.

ljvmiranda921 reviewed (approved):
LGTM! Merging later today 4pm UTC+8! Thanks @yana-xuyan for this dataloader implementation!
cc: @sabilmakbar in case there are final reviews you want to put in!
